module/VuFind/src/VuFind/Db/Service/OaiResumptionService.php

```diff
@@ -86,14 +123,24 @@ public function findToken(string $token): ?OaiResumptionEntityInterface
      */
     public function createAndPersistToken(array $params, int $expire): OaiResumptionEntityInterface
     {
-        $row = $this->createEntity()
-            ->setResumptionParameters($this->encodeParams($params))
-            ->setExpiry(\DateTime::createFromFormat('U', $expire));
-        try {
-            $this->persistEntity($row);
-        } catch (\Exception $e) {
-            $this->logError('Could not save token: ' . $e->getMessage());
-            throw $e;
+        $row = null;
+        // In extremely rare cases it might be possible that the generated random token already exists in the
+        // database. Try 5 times, but the possibility for this to happen is close to 0.
```

Comment:
To eliminate the hard-coded number in the comment, maybe this:

```suggestion
        // database. Retry up to the limit, but the possibility for this to happen is close to 0.
```